### PR TITLE
Project complete!

### PR DESCRIPTION
**Pull request by Timowest12**
We (@gracekabaghe and I ) have succesfully completed all tasks in this assignment.
In a nutshell: 

- We used the API endpoints to fetch rockets and missions.

- We dispatched the data in redux using thunk middleware.

- Added extra data to the data in our redux store to make a reservation app.

- Figured out how to merge our seperate contributions into 1 whole program.

This is our deploy link: https://deploy-preview-33--romantic-benz-43572d.netlify.app/

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
```
Comment (rloterh) [inline src/App.test.js:7]: - [x] One test case is failing. Kindly ensure to improve upon this aspect.

**Failing test case:**
image

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+} from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import MyProfile from './pages/MyProfile';
+import Missions from './components/missions/missions';
+import LOGO from './assets/planet.png';
+import Rockets from './pages/Rockets';
+import fetchRockets from './actions';
+
+function App() {
```
Comment (rloterh) [inline src/App.js]: - [x] OPTIONAL
It is a nice idea to prefer _arrow functions_ over normal functions, so as to keep your _function_ syntax simpler and cleaner.

File: src/reducer.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const initState = [];
+const rocketsReducer = (state = initState, action) => {
+  if (action.type === 'FETCH_ROCKETS') {
+    return action.payload;
+  }
+  if (action.type === 'RESERVE_ROCKET') {
+    state.map((elem) => {
+      if (elem.id === action.payload) {
+        elem.reserved = true;// eslint-disable-line no-param-reassign
```
Comment (rloterh) [inline src/reducer.js]: - [x] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is quality and clean.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Grace & Timo, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- Redux correctly implemented in state management
- Promise feature is well implemented
- Navigation is working correctly
- Application is properly deployed
- App is running as intended
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 4 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/reducer.js (hunk 3 above)
Comment (Timowest12) [reply src/reducer.js]: Removed eslinter and fixed issues

File: src/App.js (hunk 2 above)
Comment (Timowest12) [reply src/App.js]: refractored code to use arrow code instead

File: src/App.test.js (hunk 1 above)
Comment (Timowest12) [reply src/App.test.js:7]: fixed test function, all working correctly now

**Review by rukundoeric — APPROVED** [review]
Hi @Timowest12,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**